src/index.js

```diff
@@ -29,6 +29,34 @@ module.exports = () => ({
 
       _.set(path, 'node.specifiers', _.without(specifiers, ...specifiersForRemoval))
     },
+
+    ExportNamedDeclaration: (path, { opts }) => {
+      // If this is not a direct import/export then it will be handled
+      // by the ImportDeclaration visitor
+      if (path.node.source === null) {
+        if (path.node.declaration === null && _.get(path.node, 'specifiers.length') === 0) {
+          path.remove()
+        }
+
+        return
```

Comment:
I want to remove this condition there, I'm think that it's responsibility of `removeReferences` to handle such cases.  There should be similar approach for arrays and objects, BTW.